etna/lib/etna/directed_graph.rb

```diff
@@ -52,7 +52,12 @@ def as_normalized_hash(root, include_root = true)
             result[grandparent] << child_node if result.include?(grandparent)
           end
 
-          result[child_node] = []
+          # Depending on the graph shape, diamonds could lead to
+          #   resetting of previously calculated dependencies.
+          # Here we avoid resetting existing entries in `result`
+          #   and instead concatenate them if they already exist.
+          result[child_node] = [] unless result.include?(child_node)
+          result[n].concat(result[child_node]) if result.include?(child_node) && result.include?(n)
```

Comment:
Updated the path dependencies algorithm to account for the new shape of the UMAP CWL, otherwise it was getting stuck on "Confrm records" because it was not flagged as a depencendies of the parse_record_selections step.